### DISCUSSION TITLE
How to get downstream assets of an asset?

### DISCUSSION THREAD

**Discussion by bendnorman** [dagster-io/dagster#12760] [Q&A] [answered]
What is the best way to get dependency information for an asset programmatically? Which assets depend on a given asset? Looks like the `DependencyStructure` class might contain this information.

**Comment by bendnorman** [comment]
Oh I guess I can use `AssetSelection.downstream()` and `AssetSelection.upstream()`. These methods return sets of assets. Is it possible to get a data structure that describes the asset graph instead of just a flat set of assets? Also, I can't find [`AssetSelection`](https://github.com/dagster-io/dagster/blob/6a1696bb85d69fa98a7d4d1ca93f7db55aafec8a/python_modules/dagster/dagster/_core/definitions/asset_selection.py#L17) in the documentation.

**Reply by dpeng817** [reply, marked as answer]
Asset selection docs: https://docs.dagster.io/_apidocs/assets#dagster.AssetSelection
